esx_service/vmdk_ops.py

```diff
@@ -648,7 +653,9 @@ def disk_attach(vmdk_path, vm):
                      if type(dev) == vim.VirtualDisk and dev.controllerKey ==
                      controller_key])
         # search in 15 slots, with unit_number 7 reserved for scsi controller
-        availSlots = set(range(0, 6) + range(8, 16)) - taken
+        range1 = set(range(0, 7))
```

Comment:
Python 3.5 doesn't allow '+' in a set definition:

Python 3.5.1+ (default, Apr  4 2016, 12:10:58)
[GCC 4.6.3] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> y=set(range(1,5) + range(10,15))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'range' and 'range'


Works perfect in 2.7,
Python 2.7.6 (default, Mar 22 2014, 22:59:56)
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> y=set(range(1,5) + range(10,15))
>>> print y
set([1, 2, 3, 4, 10, 11, 12, 13, 14])
>>>


On Tue, Jun 28, 2016 at 3:19 AM, Mark Sterin <notifications@github.com>
wrote:

> In esx_service/vmdk_ops.py
> <https://github.com/vmware/docker-volume-vsphere/pull/479#discussion_r68662806>
> :
>
> > @@ -648,7 +653,9 @@ def disk_attach(vmdk_path, vm):
> >                       if type(dev) == vim.VirtualDisk and dev.controllerKey ==
> >                       controller_key])
> >          # search in 15 slots, with unit_number 7 reserved for scsi controller
> > -        availSlots = set(range(0, 6) + range(8, 16)) - taken
> > +        range1 = set(range(0, 7))
>
> why do we need 3 lines instad of one now ? I understand you fixed a bug
> (Missing slot #6
> <https://github.com/vmware/docker-volume-vsphere/issues/6>), but why not
> simply leave it as
>
> availSlots = set(range(0, 7) + range(8, 16)) - taken
>
> ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/479/files/618f13068273be9c4251cba120e77e704c061ed4#r68662806>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHseFMt0FppY3eb9jvekbPcqS97DAccks5qQEV0gaJpZM4I4qrt>
> .
>